src/cpu/flexcpu/inflight_inst.cc

```diff
@@ -56,6 +56,11 @@ InflightInst::InflightInst(ThreadContext* backing_context,
     staticInst(inst_ref);
 }
 
+InflightInst::~InflightInst()
+{
+    if (_traceData) delete _traceData;
```

Comment:
Why do you have this if you also have deleteTraceData?

Reply:
For things like faults that discard the InflightInst, I want to make sure it's cleared. deleteTraceData is so I can clear it earlier if we know we won't need it anymore I guess

Reply:
I see. I would pick one or the other. Can we just have the one in the destructor?

Reply:
done